evm_loader/shared-lib/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+
+#[export_root_module]
+pub fn get_library() -> NeonLib_Ref {
+    MODULE
+}
+
+#[sabi_extern_fn]
+fn hash() -> RString {
+    "".into()
+}
+
+#[sabi_extern_fn]
+fn get_version() -> RString {
+    "".into()
+}
+
+#[sabi_extern_fn]
+fn init_config(params: RStr) -> RResult<BoxedConfig<'static>, RString> {
+    fn internal(params: &str) -> Result<Config, NeonError> {
+        let api_config = serde_json::from_str(params).map_err(|_| params_to_neon_error(params))?;
+        create_from_api_comnfig(&api_config)
+    }
+    internal(params.as_str())
+        .map(DynTrait::from_value)
+        .map_err(neon_error_to_rstring)
+        .into()
+}
+
+#[sabi_extern_fn]
+fn init_context(config: &BoxedConfig, params: RStr) -> RResult<BoxedContext<'static>, RString> {
+    fn internal(config: &Config, params: &str) -> Result<Context, NeonError> {
+        let slot = serde_json::from_str(params).map_err(|_| params_to_neon_error(params))?;
+        let (rpc_client, blocking_rpc_client) = build_rpc_client(config, slot)?;
+        let signer = NeonSigner::new(config)?;
+        Ok(neon_lib::context::create(
+            rpc_client,
+            signer,
+            blocking_rpc_client,
+        ))
+    }
+    internal(config.downcast_as().unwrap(), params.as_str())
+        .map(DynTrait::from_value)
+        .map_err(neon_error_to_rstring)
+        .into()
+}
+
+#[sabi_extern_fn]
+fn init_hash_context<'a>(
+    config: &'a BoxedConfig,
+    params: RStr<'a>,
+) -> BorrowingFfiFuture<'a, RResult<BoxedContext<'static>, RString>> {
+    async fn internal(config: &Config, params: &str) -> Result<Context, NeonError> {
+        let slot = serde_json::from_str(params).map_err(|_| params_to_neon_error(params))?;
+        let (rpc_client, blocking_rpc_client) = build_hash_rpc_client(config, slot).await?;
+        let signer = NeonSigner::new(config)?;
+        Ok(neon_lib::context::create(
+            rpc_client,
+            signer,
+            blocking_rpc_client,
+        ))
+    }
+    async move {
+        internal(config.downcast_as::<Config>().unwrap(), params.as_str())
+            .await
+            .map(DynTrait::from_value)
+            .map_err(neon_error_to_rstring)
+            .into()
+    }
+    .into_ffi()
+}
+
+#[sabi_extern_fn]
+fn invoke<'a>(
+    config: &'a BoxedConfig,
+    context: &'a BoxedContext,
+    method: RStr<'a>,
+    params: RStr<'a>,
+) -> RNeonResult<'a> {
+    async move {
+        dispatch(
+            config.downcast_as().unwrap(),
+            context.downcast_as().unwrap(),
+            method.as_str(),
+            params.as_str(),
+        )
+        .await
+        .map(RString::from)
+        .map_err(neon_error_to_rstring)
+        .into()
+    }
+    .into_ffi()
+}
+
+async fn dispatch(
+    config: &Config,
+    context: &Context,
+    method: &str,
+    params: &str,
+) -> Result<String, NeonError> {
+    match method {
+        "cancel_trx" => {
+            let storage_account =
```

Comment:
Params of each method should be JSON object (map of key → value). It will help us to easily add optional arguments without breaking changes. Return value is JSON object, as well.

Reply:
The return value is already being serialized to a JSON object based on the method's return type. But okay, I'll use JSON objects for the params.